codes/typescript/chapter_hashing/hash_map.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * File: hash_map.ts
+ * Created Time: 2022-12-20
+ * Author: Daniel (better.sunjian@gmail.com)
+ */
+
+class hash_map {
+    constructor() {
+        /* 初始化哈希表 */
+        const map = new Map<number, string>();
```

Comment:
Your class is not executed. You should test the built-in Map method, so you do not need to use class.

Reply:
I've removed the `hash_map` class and only test codes are left.

Although it's not totally identical to the reference codes of Java, but definitely enough to convey the circumstances. The codes have also been tested by the below commands(I don't know what kind of test method you used, but `deno` can be easily run as `Typescript` doing so):

```shell
deno run codes/typescript/chapter_hashing/hash_map.ts
添加完成后，哈希表为
Key -> Value
Map { 12836 => "小哈", 15937 => "小啰", 16750 => "小算", 13276 => "小法", 10583 => "小鸭" }

输入学号 15937 ，查询到姓名 小啰

删除 10583 后，哈希表为
Key -> Value
Map { 12836 => "小哈", 15937 => "小啰", 16750 => "小算", 13276 => "小法" }

遍历键值对 Key->Value
12836 -> 小哈
15937 -> 小啰
16750 -> 小算
13276 -> 小法

单独遍历键 Key
12836
15937
16750
13276

单独遍历值 Value
小哈
小啰
小算
小法
```

```shell
deno run codes/typescript/chapter_hashing/array_hash_map.ts
添加完成后，哈希表为
Key -> Value
36 -> 小哈
37 -> 小啰
50 -> 小算
76 -> 小法
83 -> 小鸭

输入学号 15937 ，查询到姓名 小啰

删除 10583 后，哈希表为
Key -> Value
36 -> 小哈
37 -> 小啰
50 -> 小算
76 -> 小法

遍历键值对 Key->Value
36 -> 小哈
37 -> 小啰
50 -> 小算
76 -> 小法

单独遍历键 Key
36
37
50
76

单独遍历值 Value
小哈
小啰
小算
小法

```

Reply:
Thank you! @danielsss I see you are using deno to test but you should know that deno has skipped type-checking by default. You could see the article: https://github.com/denoland/deno/issues/11340
If you use deno to test, you need to use the --check option (deno run --check  xxxxxx.ts). Or you could use ts-node to test.

Reply:
Thank you! @danielsss I see you are using deno to test but you should know that deno has skipped type-checking by default. You could see the article: https://github.com/denoland/deno/issues/11340
If you use deno to test, you need to use the --check option 
```
 deno run --check  xxxxxx.ts
```
Or you could use ts-node to test. And then you will see the type error message, please make sure you could pass it.